File: topK_frequent_elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();

        for(int i = 0; i < nums.length; i++) {
            int number = nums[i];
            print(number)
            if(!map.containsKey(number)) {
                map.put(number, 1);
            } else {
                map.put(number, map.get(number) + 1);
            }
        }

        int[] topFrequent = new int[k];

        for(int j = 0; j < k; j++) {
            int maxKey = Collections.max(map.entrySet(), Map.Entry.comparingByValue()).getKey();
            topFrequent[j] = maxKey;
            map.remove(maxKey);
        }
        return topFrequent;
    }
}

File: top_k_frequent_elements.py
class Solution(object):
    def topKFrequent(self, nums, k):
        n = len(nums)
        m = {}
        
        if n < 2:
            return nums
        
        for i in range(n):
            if nums[i] in m:
                m[nums[i]] += 1
            else:
                m[nums[i]] = 1

        res = []
        
        for i in range(0, k):
            mx = max(m, key=m.get)
            if mx not in res:
                res.append(mx)
            del m[mx]

        
        return res
